src/ngraph/runtime/cpu/builder/avg_pool.cpp

```diff
@@ -130,23 +124,23 @@ namespace ngraph
                 if (runtime::cpu::mkldnn_utils::use_mkldnn_kernel(node))
                 {
                     auto& mkldnn_emitter = external_function->get_mkldnn_emitter();
-                    auto diff_dst_desc = runtime::cpu::mkldnn_utils::get_input_mkldnn_md(node, 0);
-                    auto diff_src_desc = runtime::cpu::mkldnn_utils::get_output_mkldnn_md(node, 0);
-
-                    size_t avg_pool_index = mkldnn_emitter->build_pooling_backward(
-                        (apb->get_include_padding_in_avg_computation()
-                             ? mkldnn::algorithm::pooling_avg_include_padding
-                             : mkldnn::algorithm::pooling_avg_exclude_padding),
-                        diff_dst_desc,
-                        diff_src_desc,
-                        apb->get_window_movement_strides(),
-                        apb->get_window_shape(),
-                        apb->get_padding_below(),
-                        apb->get_padding_above());
-
+                    auto avg_pool_fwd_desc =
+                        mkldnn_emitter->get_avg_pooling_forward_desc<ngraph::op::AvgPoolBackprop>(
+                            node, true);
+                    auto avg_pool_desc =
+                        mkldnn_emitter->get_avg_pooling_backward_desc<ngraph::op::AvgPoolBackprop>(
+                            node);
+
+                    size_t avg_pool_index = mkldnn_emitter->primitive_init(3);
```

Comment:
How is the 'count' value determined? If there is an op->constant correspondence, maybe we could use an enum to make the code more auto-descriptive and group all the values together.

Reply:
@ayzhuang i guess can we add utility function for getting the count of the number of primitives_desc? . we know how many primitive desc we need in the `get_<mkldnn>_desc()` . can we maintain another private data structure  `m_primitive_desc_count` . which uniquely index to primitive and its `count of number of desc` ?.

Reply:
Yes, we could add a utility function to return the number of primitives needed for each op based on op type. I don't think we need another private data structure since this number is only needed once.